reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -859,7 +863,25 @@ private long createNewNodeId() {
      * @return the OSM turn relation, <code>null</code>, if unsupported turn relation
      */
     OSMTurnRelation createTurnRelation(ReaderRelation relation) {
-        OSMTurnRelation.Type type = OSMTurnRelation.Type.getRestrictionType(relation.getTag("restriction"));
+        String tagRestriction = null;
+        String vehicleTypeRestricted = null;
+        List<String> vehicleTypesExcept = null;
+        if (relation.hasTag("restriction")) {
+            tagRestriction = relation.getTag("restriction");
+        } else if (relation.hasTagStartsWith("restriction:")) {
+            vehicleTypeRestricted = relation.getKeyStartsWith("restriction:");
+            if (!Helper.isEmpty(vehicleTypeRestricted)) {
```

Comment:
how can `vehicleTypeRestricted` be empty here (it is supposed to start with "restriction:" already.

Reply:
It's right, I remove the test.